utils/timeout.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2019 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
```

Comment:
Let's call it `timer` rather than `timeout`. Also, it is explicitly an implementation of [one-shot timer](https://www.freertos.org/One-shot-Vs-auto-reload-real-time-software-timers.html).

Reply:
Got it